cmd/object-handlers.go

```diff
@@ -964,14 +974,21 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 	}
 	isCpy := true
 	getRetPerms := checkRequestAuthType(ctx, r, policy.GetObjectRetentionAction, srcBucket, srcObject)
-	srcInfo.UserDefined = filterObjectLockMetadata(ctx, r, srcBucket, srcObject, srcInfo.UserDefined, isCpy, getRetPerms)
+	getLegalHoldPerms := checkRequestAuthType(ctx, r, policy.GetObjectLegalHoldAction, srcBucket, srcObject)
```

Comment:
Does this matter? I do not know, Please let me know if this scenario has been tested on AWS.

Reply:
yes it does - bucket needs to have object locking enabled for placing a legal hold. See example below against aws
```
➜  aws s3api put-object --body ~/data/datafile-1-kB --bucket uswest7 --key b2 --object-lock-legal-hold-status=ON     

An error occurred (InvalidRequest) when calling the PutObject operation: Bucket is missing ObjectLockConfiguration
```